### PR TITLE
Collect BOM constraints automatically

### PR DESCRIPTION
**Pull request by swankjesse**
Closes: https://github.com/square/okio/issues/1121

### CODE REVIEW

File: build-support/src/main/kotlin/bom.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+
+/**
+ * Collect all the root project's multiplatform targets and add them to the BOM.
+ *
+ * Only published subprojects are included.
+ *
+ * This supports Kotlin/Multiplatform and Kotlin/JS subprojects.
+ */
+fun Project.collectBomConstraints() {
```
Comment (swankjesse) [inline build-support/src/main/kotlin/bom.kt:18]: FYI @oldergod we might wanna copy-paste this into our other repos? Or maybe submit it as a new feature for the Vanniktech plugin? Note that I don’t yet support non-multiplatform Kotlin/JVM subprojects.

Comment (swankjesse) [reply build-support/src/main/kotlin/bom.kt:18]: Oooh, just realized I gotta support the Kotlin-Android plugin to capture the AssetFileSystem.

Comment (oldergod) [reply build-support/src/main/kotlin/bom.kt:18]: I like it very much indeed

**Review by oldergod — APPROVED** [review]

File: build-support/src/main/kotlin/bom.kt (hunk 1 above)
Comment (swankjesse) [reply build-support/src/main/kotlin/bom.kt:21]: Done.

**Pull request merged**